File: greedy/3223-minimum-length-of-string-after-operations.py
from collections import Counter
import unittest

# https://leetcode.com/problems/minimum-length-of-string-after-operations/

# python3 -m unittest greedy/3223-minimum-length-of-string-after-operations.py

class Solution(unittest.TestCase):
    def minimumLength(self, s: str) -> int:
        return len(s) - sum(cnt - 2 + (cnt&1) for cnt in Counter(s).values())

    def test(self):
        for s, expected in [
            ("abaacbcbb", 5),
            ("aa", 2),
        ]:
            output = self.minimumLength(s)
            self.assertEqual(expected, output, f"expected: {expected}, output: {output}")

File: greedy/3223-minimum-length-of-string-after-operations.go
package greedy

// https://leetcode.com/problems/minimum-length-of-string-after-operations/

// Approach: Greedy
// Time: O(n)
// Space: O(1)
func minimumLength(s string) int {
	var count [26]int
	for _, c := range s {
		count[int(c-'a')]++
	}
	length := 0
	for _, cnt := range count {
		if cnt&1 == 1 {
			length += 1
		} else if cnt != 0 {
			length += 2
		}
	}
	return length
}
